service/controller/app/resource/chart/current.go

```diff
@@ -27,14 +27,21 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 		return nil, microerror.Mask(err)
 	}
 
-	if cc.Status.TenantCluster.IsDeleting {
+	if cc.Status.ChartStatus.Status != "" {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("chart %#q failed to merge configMaps/secrets, no need to reconcile resource", cr.Name))
+		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+		resourcecanceledcontext.SetCanceled(ctx)
```

Comment:
Whenever we failed to merge configMaps/secrets, we don't need to proceed chart install, otherwise, it would result in unexpected outcomes from the customer.